### PR TITLE
Persistent graph element

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

File: analyzer/server.go [hunk 1]
```diff
@@ -339,12 +339,13 @@ func NewServerFromConfig() (*Server, error) {
 	flowSubscriberWSServer := ws.NewStructServer(config.NewWSServer(hub.HTTPServer(), "/ws/subscriber/flow", apiAuthBackend))
 	flowSubscriberEndpoint := server.NewFlowSubscriberEndpoint(flowSubscriberWSServer)
 
-	captureAPIHandler, err := api.RegisterCaptureAPI(hub.APIServer(), g, apiAuthBackend)
+	apiServer := hub.APIServer()
```
Comment (lebauce) [inline analyzer/server.go:342]: Nice catch :-)

File: graffiti/hub/hub.go [hunk 2]
```diff
@@ -174,50 +227,58 @@ func NewHub(id string, serviceType service.Type, listen string, g *graph.Graph,
...
-	}
+	hub.httpServer = httpServer
+	hub.apiServer = apiServer
+	hub.podWSServer = podWSServer
+	hub.replicationEndpoint = replicationEndpoint
+	hub.replicationWSServer = replicationWSServer
+	hub.publisherWSServer = publisherWSServer
+	hub.subscriberWSServer = subscriberWSServer
+	hub.traversalParser = tr
+	hub.etcdClient = opts.EtcdClient
```
Comment (lebauce) [inline graffiti/hub/hub.go:281]: nit: this line could be moved to Hub instantiation

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

File: agent/agent.go [hunk 3]
```diff
@@ -121,7 +121,8 @@ func NewAgent() (*Agent, error) {
 	}
 
 	hostID := config.GetString("host_id")
-	g := graph.NewGraph(hostID, backend, config.AgentService)
+	origin := fmt.Sprintf("%s.%s", config.AgentService, hostID)
```
Comment (lebauce) [inline agent/agent.go:124]: nit: common.Origin(config.AgentService, hostID)

File: graffiti/common/origin.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (lebauce) [inline graffiti/common/origin.go:1]: What about moving this file to the `graph` folder ?

**Comment by lebauce** [comment]
run unit-tests

**Comment by lebauce** [comment]
run unit-tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by lebauce** [comment]
run k8s-tests
run functional-tests-backend-orientdb
run compile-tests
run cdd-overview-tests

**Comment by lebauce** [comment]
run cdd-overview-tests

**Comment by lebauce** [comment]
run k8s-tests

**Comment by lebauce** [comment]
run k8s-tests

**Comment by lebauce** [comment]
run k8s-tests

**Comment by lebauce** [comment]
run tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by safchain** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run scale-tests

**Comment by nplanel** [comment]
run scale-tests

File: cmd/client/edge.go [hunk 5]
```diff
@@ -219,10 +219,12 @@ var EdgeUpdate = &cobra.Command{
 }
 
 func addCreateEdgeFlags(cmd *cobra.Command) {
+	host, _ = os.Hostname()
 	cmd.Flags().StringVarP(&edgeType, "edge-type", "", "", "edge type")
 	cmd.Flags().StringVarP(&parentNodeID, "parent", "", "", "parent node identifier")
 	cmd.Flags().StringVarP(&childNodeID, "child", "", "", "child node identifier")
 	cmd.Flags().StringVarP(&metadata, "metadata", "", "", "edge metadata, key value pairs. 'k1=v1, k2=v2'")
+	cmd.Flags().StringVarP(&metadata, "host", "", host, "host")
```
Comment (nplanel) [inline cmd/client/edge.go:227]: cmd.Flags().StringVarP(&host, "host", "", host, "host")

File: graffiti/storage/elasticsearch/client.go [hunk 6]
```diff
@@ -131,6 +134,25 @@ func (c *Client) addMapping(index Index) error {
...
 
+func (c *Client) checkIndices() error {
+	aliases, err := c.esClient.Aliases().Do(context.Background())
+	if err != nil {
+		return err
+	}
+
+LOOP:
+	for _, index := range c.indices {
+		for name := range aliases.Indices {
```
Comment (nplanel) [inline graffiti/storage/elasticsearch/client.go:145]: may create a temp map with aliases.Indices and lookup index if exist or not

**Pull request merged**